### DISCUSSION TITLE
Early Stopping for TimeSeries

### DISCUSSION THREAD

**Discussion by mreismendes** [autogluon/autogluon#3212] [Q&A]
Is it possible to set Early Stopping when running Time Series Forecasting?
Thank you

**Comment by shchur** [comment]
Currently early stopping for GluonTS models (DeepAR, TFT, SimpleFeedforward) is done "implicitly", as described [here](https://github.com/awslabs/gluonts/issues/1184#issuecomment-771436931).

You could add an early stopping callback using the `hyperparameters` keyword:

```python
from lightning.pytorch.callbacks.early_stopping import EarlyStopping

predictor = TimeSeriesPredictor(...).fit(
    train_data, 
    hyperparameters={"DeepAR": {"callbacks": [EarlyStopping(monitor="val_loss")]},
)
```

We plan to enable this by default in a future release.

**Comment by mreismendes** [comment]
Hi @shchur,

Thanks for your reply to my question about early stopping for time series forecasting. I tried to implement your suggestion, but I'm still having trouble.

I'm not sure what I'm doing wrong. Can you help me troubleshoot this?

Thanks,

**Reply by shchur** [reply]
Could you show me what error message you're receiving & share the code that reproduces this problem?
